scala-thoughts.md

```diff
@@ -0,0 +1,5 @@
+In theory, I think, it's pretty easy to do in Scala anything that's easy to do in Java. Some of that may not be "good" Scala, though. Basic object-oriented programming, for example isn't any more difficult (even if the idea of companion objects is kind of weird at first). Functional programming is pretty easy in Scala as well, although I was surprised to find that higher-order functions are typically methods of types for which those functions make any sense. I found that it was
+quite difficult to use recursion and pattern-matching to do something better handled without the pattern-matching. In particular, I basically gave up figuring out how to implement zipMultiple with pattern-matching, looked at the sample solution, realized _why_ I hadn't figured it out (it was considerably more complicated than I think is necessary), and ended up just referring to Prof. Ben's solution. Obviously there are going to be some tasks that pattern matching won't be suited for regardless of the
```

Comment:
I agree completely that the pattern matching recursion examples were needlessly complex for the sake of instruction. I honestly had a really hard time doing them with pattern matching without the pattern matching being incredibly trivial.